src/js/tech/html5.js

```diff
@@ -209,11 +209,40 @@ class Html5 extends Tech {
   }
 
   /**
-   * Handle IOS Headphone disconnection during playback
+   * Handle IOS Headphone disconnection during playback after successful
+   * playback of each video source.
    *
    * @private
   */
   handleIOSHeadphonesDisconnection_() {
+    // try to determine whether a successful playback occurred before
+    // checking stalled and suspend events
+    const SUCCESSFUL_PLAYBACK_TIME_SECONDS = 1;
+    const boundStalledAndSuspendHandler = this.handleIOSHeadphoneStalledAndSuspendEvent_.bind(this);
+
+    const timeUpdateListener = function() {
+      if (this.currentTime() > SUCCESSFUL_PLAYBACK_TIME_SECONDS) {
+        this.off('timeupdate', timeUpdateListener);
+
+        // If iOS check if we have a real stalled or supend event or
+        // we got stalled/suspend due headphones where disconnected during playback
+        this.on('stalled', boundStalledAndSuspendHandler);
```

Comment:
this can be combined into one line by using an array of values `['stalled', 'suspend']`, can also be used for the `.off` below.

Reply:
Adjusted the pull here https://github.com/videojs/video.js/pull/6354/commits/658dcf53931827f1a21254c61541c5a74bb58cb1